### PR TITLE
IGeoSvc: add constantAsString accessor

### PR DESCRIPTION
**Pull request by andresailer**
This lets us isolate other components from a direct DD4hep dependency

BEGINRELEASENOTES
- IGeoSvc: Add function constantAsString, to access dd4hep constants, but isolating other components from DD4hep
- IGeoSvc: Add function getDetector as a replacement for the new deprecated lcdd function

ENDRELEASENOTES

This is part of addressing: key4hep/k4MarlinWrapper#27

### CODE REVIEW

File: k4Interface/include/k4Interface/IGeoSvc.h [hunk 1]
```diff
@@ -32,9 +32,10 @@ class G4VUserDetectorConstruction;
 class GAUDI_API IGeoSvc : virtual public IService {
 public:
   DeclareInterfaceID(IGeoSvc, 1, 0);
-  virtual dd4hep::DetElement           getDD4HepGeo() = 0;
-  virtual dd4hep::Detector*            lcdd()         = 0;
-  virtual G4VUserDetectorConstruction* getGeant4Geo() = 0;
+  virtual dd4hep::DetElement           getDD4HepGeo()                            = 0;
+  virtual dd4hep::Detector*            lcdd()                                    = 0;
```
Comment (tmadlener) [inline k4Interface/include/k4Interface/IGeoSvc.h:36]: Not really part of this PR but on the bigger picture, should this be named `getDetector` or something slightly less specific than `lccd`?

Comment (andresailer) [reply k4Interface/include/k4Interface/IGeoSvc.h]: Done

**Review by tmadlener — APPROVED** [review]
